test/src/test/java/org/corfudb/integration/ServerRestartIT.java

```diff
@@ -262,15 +264,30 @@ private void runSingleNodeRecoveryTransactionalClient(boolean nested) throws Exc
         // ShutDown (STOP) CORFU Server
         assertThat(shutdownCorfuServer(corfuServerProcess)).isTrue();
 
-        // Execute Transactions (once Corfu Server Shutdown)
-        for (int i = 0; i < ITERATIONS; i++) {
-            assertThat(executeTransaction(runtime, smrMapList, expectedMapList,
-                    MIN_LIMIT_KEY_RANGE_DURING_SHUTDOWN, MAX_LIMIT_KEY_RANGE_DURING_SHUTDOWN,
-                    nested, rand)).isFalse();
-        }
+        // Schedule offline transactions, first one should be stuck and will eventually succeed
+        // on reconnect
+        ScheduledThreadPoolExecutor offline = new ScheduledThreadPoolExecutor(1);
+        ScheduledFuture<Boolean> offlineTransactionsSucceeded = offline.schedule(() -> {
+            for (int i = 0; i < ITERATIONS; i++) {
+                boolean txState = executeTransaction(runtime, smrMapList, expectedMapList,
+                        MIN_LIMIT_KEY_RANGE_DURING_SHUTDOWN, MAX_LIMIT_KEY_RANGE_DURING_SHUTDOWN,
+                        nested, rand);
+
+                if (!txState) {
+                    return false;
+                };
+            }
+
+            return true;
+        },CLIENT_DELAY_POST_SHUTDOWN, TimeUnit.MILLISECONDS);
+        offline.shutdown();
+
+        Thread.sleep(CORFU_SERVER_DOWN_TIME);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this use of "Thread.sleep()". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2925)